src/hotspot/share/opto/vectornode.cpp

```diff
@@ -894,13 +947,28 @@ LoadVectorNode* LoadVectorNode::make(int opc, Node* ctl, Node* mem,
   return new LoadVectorNode(ctl, mem, adr, atyp, vt, control_dependency);
 }
 
+Node* LoadVectorNode::Ideal(PhaseGVN* phase, bool can_reshape) {
+  const TypeVect* vt = as_LoadVector()->vect_type();
```

Comment:
Why you need `as_LoadVector()` fro `this`? Same in `StoreVectorNode::Ideal().

Reply:
Good catch and thanks! We could directly use "vect_type()" here. I will change this later.